compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -734,17 +782,19 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
     val defn.ArrayOf(elemtp) = tp: @unchecked
     if (isGenericArrayElement(elemtp, isScala2 = sourceLanguage.isScala2)) defn.ObjectType
     else
-      try JavaArrayType(erasureFn(sourceLanguage, semiEraseVCs = false, isConstructor, isSymbol, wildcardOK)(elemtp))
+      try
+        val eElem = erasureFn(sourceLanguage, semiEraseVCs = false, isConstructor, isSymbol, inSigName)(elemtp)
+        if eElem == null then null
+        else JavaArrayType(eElem)
       catch case ex: Throwable =>
         handleRecursive("erase array type", tp.show, ex)
   }
 
-  private def erasePair(tp: Type)(using Context): Type = {
-    // NOTE: `tupleArity` does not consider TypeRef(EmptyTuple$) equivalent to EmptyTuple.type,
-    // we fix this for printers, but type erasure should be preserved.
-    val arity = tp.tupleArity
-    if (arity < 0) defn.ProductClass.typeRef
-    else if (arity <= Definitions.MaxTupleArity) defn.TupleType(arity).nn
+  private def erasePair(tp: Type)(using Context): Type | Null = {
+    val arity = tupleArity(tp)
+    if arity == -2 then null // erasure depends on an uninstantiated type variable or WildcardType
```

Comment:
Elsewhere we make sure that we return `null` only if `inSigName` is true. But here it seems no such check is done.

Reply:
erasePair is called from TypeErasure#apply which does the null check, elsewhere we check that our input is not null which isn't necessary here with non-nullable types.